src/scene/mesh.js

```diff
@@ -175,9 +176,9 @@ class GeometryVertexStream {
  * @property {pc.Skin} [skin] The skin data (if any) that drives skinned mesh animations for this mesh.
  * @property {pc.Morph} [morph] The morph data (if any) that drives morph target animations for this mesh.
  */
-class Mesh {
+class Mesh extends RefCount {

```

Comment:
This feels as not an "ideal" use of inheritance, as both those things are very different. It feels as reference counter, should be just a property, not a super class of Mesh.

Reply:
Or, if it serves as "destroyable object", then it probably should be something like: "DestroyableResource", with reference counting inside, then it feels as good inheritable class.

Reply:
Yeah, I wondered something similar about the naming. e.g. 'RefCountedObject' or something.

Reply:
Sorry for offtopic, but it reminded me those days where we would spend hours arguing on how to call things 😆 
Seems like it never gets better with time. 😋

Reply:
I think it's easy to see these 'arguments' (or discussions) as slightly petty. But I think they're incredibly valuable for building a consistent and intuitive API. I think we all have valuable opinions to share to help us make collectively great decisions.

Reply:
That's the intention behind it, but I didn't want all destroyable resources to have to be ref counted. In typed languages world (c#, typescript) I'd have base class with ref counting, and enforce IDestroyable and ILostContext functions using interfaces.

I also considered hierarchy with top level being Resource, which would have Destroy and LostContext functions, next level would be RefCountedResource which would add ref counter. Then engine classes could extend either of these.

At the moment recently I added loseContext to bunch of classes to handle that case, but I'd really like an interface in JS. Maybe TS?